src/feature/dirauth/process_descs.c

```diff
@@ -175,12 +216,15 @@ dirserv_load_fingerprint_file(void)
 
   for (list=front; list; list=list->next) {
     char digest_tmp[DIGEST_LEN];
+    char digest256_tmp[DIGEST256_LEN];
     router_status_t add_status = 0;
     nickname = list->key; fingerprint = list->value;
     tor_strstrip(fingerprint, " "); /* remove spaces */
-    if (strlen(fingerprint) != HEX_DIGEST_LEN ||
+    int is_ed25519 = (strlen(fingerprint) == BASE64_DIGEST256_LEN);
+    if ((strlen(fingerprint) != HEX_DIGEST_LEN ||
```

Comment:
Hmm, this `if` check got substatntially harder to understand, any chance of disentangling into multiple statements or making it simpler in some way?